src/devices/emax.c

```diff
@@ -0,0 +1,233 @@
+- R: (16) Rain
+- W: (12) Wind speed
+- D: (9 bit) Wind Direction
+- U: (4 bit) UV index
+- L: (1 + 15 bit) Lux value, if first bit = 1 , then x 10 the rest.
+- A: (4 bit) fixed values of 0xA
+- 0: (4 bit) fixed values of 0x0
+- xx: incremental value each tx
+- yy: incremental value each tx yy = xx + 1
+- S: (8 bit) checksum
+
+Raw Data: 
+
+    ff ff 80 00 aa aa aa aa aa ca ca 54
+    aa 04 59 41 06 1f 42 01 01 01 81 01 16 01 01 01 04 05 06 07 08 09 10 11 12 13 14 15 16 17 9d ad 9e
+    0000
+
+Format string:
+
+    8h K:4h CH:4h ID:12h Flags:4b 4h Temp:12h Hum:8h 4h Wind:12h 4h Direction: 12h Rain: 16h 4h UV:4h Lux:16h  112h xx:8d CHKSUM:8h
+
+Decoded example:
+
+    aa KD:0 CH:4 ID:594 FLAGS:0001 0 TEMP:61f (66.7F) HUM:42 (66%) Wind: 101 ( = 000 * 0.2 = 0 kmh) 0 Direction: 181 ( = 080 = 128°) Rain: 0116 ( 0015 * 0.2  = 4.2 mm) 0 UV: 1 (0 UV) Lux: 0101 (0 Lux) 04 05 ...16 17 xx:9d CHKSUM:ad yy:9e
+    
+*/
+
+static int emax_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    // full preamble is ffffaaaaaaaaaacaca54
+    uint8_t const preamble_pattern[] = {0xaa, 0xaa, 0xca, 0xca, 0x54};
+
+    int ret = 0;
+    for (unsigned row = 0; row < bitbuffer->num_rows; ++row) {
+        unsigned pos = bitbuffer_search(bitbuffer, row, 0, preamble_pattern, sizeof(preamble_pattern) * 8);
+
+        if (pos >= bitbuffer->bits_per_row[row]) {
+            decoder_log(decoder, 2, __func__, "Preamble not found");
+            ret = DECODE_ABORT_EARLY;
+            continue;
+        }
+        decoder_logf(decoder, 2, __func__, "Found row: %d", row);
+
+        pos += sizeof(preamble_pattern) * 8;
+        // we expect 32 bytes
+        if (pos + 32 * 8 > bitbuffer->bits_per_row[row]) {
+            decoder_log(decoder, 2, __func__, "Length check fail");
+            ret = DECODE_ABORT_LENGTH;
+            continue;
+        }
+        uint8_t b[32] = {0};
+        bitbuffer_extract_bytes(bitbuffer, row, pos, b, sizeof(b) * 8);
+
+        // verify checksum
+        if ((add_bytes(b, 31) & 0xff) != b[31]) {
+            decoder_log(decoder, 2, __func__, "Checksum fail");
+            ret = DECODE_FAIL_MIC;
+            continue;
+        }
+
+        int channel     = (b[1] & 0x0f);
+        int kind        = ((b[1] & 0xf0) >> 4); 
+        int id          = (b[2] << 4) | (b[3] >> 4);
+        int battery_low = (b[3] & 0x08);
+        int pairing     = (b[3] & 0x04);
+        
+        // depend if external temp/hum sensor or Weather rain/wind station the values are not decode the same
+        
+        if (kind != 0) {  // if not Rain/Wind ... sensor
+        
+            int temp_raw    = ((b[4] & 0x0f) << 8) | (b[5] & 0xf0) | (b[6] & 0x0f); // weird format
+            float temp_f    = (temp_raw - 900) * 0.1f;
+            int humidity    = b[7];
+
+            /* clang-format off */
+            data_t *data = data_make(
+                    "model",            "",                 DATA_STRING, "EMAX Temperature Humidity sensor",
```

Comment:
We can not change this now. Needs to stay "Altronics-X7064".
Change this later in new PR, e.g. "Altronics-X7064" -> "Emax-EM3390W6"
(see Model at https://triq.org/rtl_433/DATA_FORMAT.html#message-data)